workflow/rules/htsinfer.smk

```diff
@@ -0,0 +1,101 @@
+"""Independent Snakefile for inferring sample specific parameters with HTSinfer."""
+
+import os
+import pandas as pd
+
+# set config defaults if not given
+config.setdefault("records", 100000)
+
+
+# global variables
+samples = pd.read_csv(config["samples"], header=0, index_col=0, sep = "\t")
+OUT_DIR = config["outdir"]
+LOG_DIR = os.path.join(OUT_DIR,"logs")
+CLUSTER_LOG = os.path.join(LOG_DIR,"cluster_logs")
+# Write inferred params into new sample table.
+SAMPLES_OUT = os.path.join(OUT_DIR,config["samples_out"])
+
+
+
+localrules: all, htsinfer_to_tsv
+
+rule all:
+    input:
+        SAMPLES_OUT
+
+current_rule = "run_htsinfer"
+rule run_htsinfer:
+    ''' Run htsinfer on fastq samples    
+    '''
+    input:
+        fq1_path = lambda wildcards:
+                samples.loc[wildcards.sample, "fq1"]
+    output:
+        htsinfer_json = os.path.join(OUT_DIR, "htsinfer_{sample}.json")
+    params:
+        fq2_path = lambda wildcards: 
+                (samples.loc[wildcards.sample,"fq2"] if samples.loc[wildcards.sample,"fq2"] else ""),
+        records = config["records"],
+        outdir = OUT_DIR,
+        cluster_log_path = CLUSTER_LOG
+    threads: 4
+    singularity: 
+        "docker://zavolab/htsinfer:0.9.0"
+    conda: 
+        os.path.join(workflow.basedir, "envs", "htsinfer.yaml")
```

Comment:
this current path does not work, it points to `./workflow/rules/envs/htsinfer.yaml` but the file is in `./workflow/envs/`.
should be something like: `os.path.join(workflow.basedir, "..", "envs", "htsinfer.yaml")` (if you want to follow the same architecture as `sra_download.smk`.)
Same applies to other rule.